docs/react.md

```diff
@@ -0,0 +1,48 @@
+# React Guidelines
+
+## Purpose
+
+This document provides guidelines for contributing new components or updating existing components in our React codebases.
+
+## Guidelines for React Components
+
+The following guidelines are not strictly required, but are considered best practices that contribute to higher quality components. Not all guidelines will be applicable depending on the specific React repository's support. Please refer to the respective repository for additional guidelines and further details.
+
+If certain requirements cannot be met due to time constraints, legacy code, or other factors, contributors should use their best judgement on which requirements are most important or feasible to implement.
+
+### Component Structure
+
+- Follow consistent file structure
+- Use functional components with hooks instead of class components
+- Separate container and presentational components when appropriate
+- Avoid large, complex components - break into reusable pieces
+
+### Component User Interface Features (If applicable)
+
+- Accessibility (A11y)
+- Dark-mode
+- Dynamic text length
+- Internationalization (i18n) translations including right-to-left languages
+- Responsive UI/UX (support mobile, fullscreen, and pop-up views)
```

Comment:
Could we add use of design tokens as opposed to static color and typography values? This will be critical in our brand evolution strategy as well as maintain consistency and accessibility standards

```suggestion
- Responsive UI/UX (support mobile, fullscreen, and pop-up views)
- Use of [design tokens](https://github.com/MetaMask/design-tokens) for color and typography to ensure brand consistency and uphold accessibility standards
```

Reply:
absolutely! Thanks @georgewrmarshall! added [bc41f20](https://github.com/MetaMask/contributor-docs/pull/65/commits/bc41f20a9b24b0564f65646e74d678ba8ff71c43)